components/overflow-group/overflow-group.js

```diff
@@ -0,0 +1,495 @@
+		this._overflowMenuMini = this.shadowRoot.querySelector('.d2l-overflow-dropdown-mini');
+		if (this.openerType === OPENER_TYPE.ICON && this._overflowMenuMini) {
+			this._overflowMenuWidth = this._overflowMenuMini.offsetWidth;
+		} else if (this._overflowMenu) {
+			this._overflowMenuWidth = this._overflowMenu.offsetWidth;
+		}
+
+		const showing = {
+			count: 0,
+			width: 0
+		};
+
+		let isSoftOverflowing, isForcedOverflowing;
+		for (let i = 0; i < this._layoutItems.length; i++) {
+			const itemLayout = this._layoutItems[i];
+
+			// handle minimum items to show
+			if (showing.count < this.minToShow) {
+				showing.width += itemLayout.width;
+				showing.count += 1;
+				itemLayout.trigger = 'force-show';
+				itemLayout.isChomped = false;
+				continue;
+			}
+
+			// handle maximum items to show
+			if (this.maxToShow >= 0 && showing.count >= this.maxToShow) {
+				isForcedOverflowing = true;
+				itemLayout.isChomped = true;
+				itemLayout.trigger = 'force-hide';
+				continue;
+			}
+
+			// chomp or unchomp based on space available, and we've already handled min/max above
+			if (!isSoftOverflowing && showing.width + itemLayout.width < this._availableWidth) {
+				showing.width += itemLayout.width;
+				showing.count += 1;
+				itemLayout.isChomped = false;
+				itemLayout.trigger = 'soft-show';
+
+			} else {
+				isSoftOverflowing = true;
+				itemLayout.isChomped = true;
+				itemLayout.trigger = 'soft-hide';
+
+			}
+
+		}
+		// if there is at least one showing and no more to be hidden, enable collapsing more button to [...]
+		this._overflowMenuHidden = items.length === showing.count;
+		if (!this._overflowMenuHidden && (isSoftOverflowing || isForcedOverflowing)) {
+			for (let j = this._layoutItems.length; j--;) {
+				if (showing.width + this._overflowMenuWidth < this._availableWidth) {
+					break;
+				}
+				const itemLayoutOverflowing = this._layoutItems[j];
+				if (itemLayoutOverflowing.trigger !== 'soft-show') {
+					continue;
+				}
+				showing.width -= itemLayoutOverflowing.width;
+				showing.count -= 1;
+				isSoftOverflowing = true;
+				itemLayoutOverflowing.trigger = 'soft-hide';
+				itemLayoutOverflowing.isChomped = true;
+			}
+		}
+		const overflowDropdownOverflowing = (showing.width + this._overflowMenuWidth >= this._availableWidth);
+		const swapToMiniButton = overflowDropdownOverflowing && !this._overflowMenuHidden;
+
+		this._mini = this.openerType === OPENER_TYPE.ICON || swapToMiniButton;
+		this._chompIndex = this._overflowMenuHidden ? null : showing.count;
+
+		this.dispatchEvent(new CustomEvent('d2l-overflow-group-updated', { bubbles: true, composed: true }));
+	}
+	_convertToDropdownItem(node) {
+		const tagName = node.tagName.toLowerCase();
+		let menuItem;
+		if (tagName === 'd2l-button' || tagName === 'd2l-button-subtle' || tagName === 'button' || tagName === 'd2l-button-icon') {
+			menuItem = this._createMenuItem(node);
```

Comment:
Should be able to just `return` from each of these blocks, then remove the `} else { return; } return menuItem;` at the end. A `switch` on the `tagName` may also look a little cleaner, and then handle the "separator" and "custom-item" case separately (which may go away anyway).